src/Compilers/CSharp/Test/Syntax/Parsing/TopLevelStatementsParsingTests.cs

```diff
@@ -2626,33 +2714,902 @@ public void UsingAliasTest()
+                                {
+                                    N(SyntaxKind.IdentifierToken, "WriteLine");
+                                }
+                            }
+                            N(SyntaxKind.ArgumentList);
+                            {
+                                N(SyntaxKind.OpenParenToken);
+                                N(SyntaxKind.CloseParenToken);
+                            }
+                        }
+                        N(SyntaxKind.SemicolonToken);
+                    }
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void ErrorRecovery_04()
+        {
+            var test = @"extern alias ";
+
+            UsingTree(test,
+                // (1,8): error CS0116: A namespace cannot directly contain members such as fields, methods or statements
+                // extern alias 
+                Diagnostic(ErrorCode.ERR_NamespaceUnexpected, "alias").WithLocation(1, 8)
+                );
+
+            N(SyntaxKind.CompilationUnit);
+            {
+                N(SyntaxKind.IncompleteMember);
+                {
+                    N(SyntaxKind.ExternKeyword);
+                    N(SyntaxKind.IdentifierName);
+                    {
+                        N(SyntaxKind.IdentifierToken, "alias");
+                    }
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void ErrorRecovery_05()
+        {
+            var test = @"using aliasY = X.Y;";
+
+            UsingTree(test);
+
+            N(SyntaxKind.CompilationUnit);
+            {
+                N(SyntaxKind.UsingDirective);
+                {
+                    N(SyntaxKind.UsingKeyword);
+                    N(SyntaxKind.NameEquals);
+                    {
+                        N(SyntaxKind.IdentifierName);
+                        {
+                            N(SyntaxKind.IdentifierToken, "aliasY");
+                        }
+                        N(SyntaxKind.EqualsToken);
+                    }
+                    N(SyntaxKind.QualifiedName);
+                    {
+                        N(SyntaxKind.IdentifierName);
+                        {
+                            N(SyntaxKind.IdentifierToken, "X");
+                        }
+                        N(SyntaxKind.DotToken);
+                        N(SyntaxKind.IdentifierName);
+                        {
+                            N(SyntaxKind.IdentifierToken, "Y");
+                        }
+                    }
+                    N(SyntaxKind.SemicolonToken);
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void ErrorRecovery_06()
+        {
+            var test = @"
+using X;
+using aliasY = X.Y;
```

Comment:
This one too. I'm not sure why these tests are named `ErrorRecovery_X`.

Reply:
Same response as for ```ErrorRecovery_05```. The tests are testing code responsible for an error recovery. Both when it should and shouldn't occur.